### PR TITLE
Fix add/delete user to group 0 via provisioning API

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
1) Add scenarios for adding users to group 0.
2) Add scenarios for deleting users from group 0.
Those scenarios fail, so fix the code:
3) When checking the group id, do not test with the ``empty()`` function - test explicitly for the things that are "empty" apart from ``0``.

## Related Issue
#29990 

## Motivation and Context
Remove more uses of ``empty()`` that cause problems for data that has the valid value "0".

## How Has This Been Tested?
Run the API tests locally.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: apps/provisioning_api/lib/Users.php [hunk 1]
```diff
@@ -475,8 +475,8 @@ public function addToGroup($parameters) {
 			return new Result(null, API::RESPOND_UNAUTHORISED);
 		}
 
-		$groupId = !empty($_POST['groupid']) ? $_POST['groupid'] : null;
-		if($groupId === null) {
+		$groupId = $_POST['groupid'];
+		if (($groupId === '') || ($groupId === null) || ($groupId === false)) {
```
Comment (phil-davis) [inline apps/provisioning_api/lib/Users.php]: Note: this uses the same style of check that was done in Groups.php ``addGroup`` a while ago.

**Comment by phil-davis** [comment]
Note: the 2nd commit is needed because the test scenarios are not independent. The scenarios do not seem to clean up their state (in this case which users are in which groups) at the end of each scenario.

As a separate issue, that should all be sorted out some day.

**Comment by individual-it** [comment]
Looks good, should we add tests that test really empty group names?

**Comment by phil-davis** [comment]
This scenario seems to send an empty groupid:
https://github.com/owncloud/core/blob/a3443144274101d1f0944de1f4b5bf1e252380b7/tests/integration/features/provisioning-v1.feature#L93

**Review by individual-it — APPROVED** [review]

**Comment by phil-davis** [comment]
Backport stable10 #30004
